EnhancedMediaPlayer/MediaPlayerViewController.swift

```diff
@@ -55,17 +78,85 @@ public class MediaPlayerViewController: UIHostingController<MediaPlayerView> {
     }
     
     private func rewind() {
-        // TODO: implement rewind
+        seek(forward: false)
     }
     
     private func forward() {
-        // TODO: implement forward
+        seek(forward: true)
+    }
+
+    private func seek(forward: Bool) {
+        let current = viewModel.player.currentTime().seconds
+        let target: Double = forward ? current + viewModel.seekFactor : current - viewModel.seekFactor
+        let result: CMTime =  .init(seconds: target, preferredTimescale: Constants.timeScale)
+        viewModel.player.seek(to: result)
     }
     
     private func onLoad() {
         play()
         observeMediaFinished()
     }
+
+    private func onTapSeekArea(_ seek: MediaPlayerTappableView.Seek) {
+        guard self.viewModel.isSeekByTapEnabled else {
+            viewModel.areControlsVisible.toggle()
+            return
+        }
+
+        self.tapTimer?.invalidate()
+        self.seekAnimationTimer?.invalidate()
+        self.viewModel.visibleSeek = nil
+
+        switch viewModel.seekState {
+        case .none:
+            viewModel.seekState = .ready(seek)
+            resetToDefaultSeekState()
+        case .ready:
+            viewModel.seekState = .seeking(seek)
+            fallthrough
+        case .seeking:
+            performSeek(seek)
+            terminateSeek()
+            runSeekAnimation(seek)
+        }
+    }
+
+    private func performSeek(_ seek: MediaPlayerTappableView.Seek) {
+        switch seek {
+        case .backward:
+            self.rewind()
+        case .forward:
+            self.forward()
+        }
+    }
+
+    private func terminateSeek() {
+        self.tapTimer = Timer.scheduledTimer(withTimeInterval: Constants.seekTapInterval, repeats: false) { _ in
+            self.viewModel.seekState = .none
+        }
+    }
+
+    private func resetToDefaultSeekState() {
+        self.tapTimer = Timer.scheduledTimer(withTimeInterval: Constants.showControlsDelay, repeats: false) { _ in
+            guard case .ready = self.viewModel.seekState else { return }
+
+            self.viewModel.areControlsVisible.toggle()
+            self.viewModel.seekState = .none
+        }
+    }
+
+    private func runSeekAnimation(_ seek: MediaPlayerTappableView.Seek) {
+        self.viewModel.visibleSeek = seek
+
+        self.seekAnimationTimer = Timer.scheduledTimer(
+            withTimeInterval: Constants.seekTapAnimationInterval,
+            repeats: false
+        ) { _ in
+            withAnimation {
+                self.viewModel.visibleSeek = nil
+            }
+        }
+    }
```

Comment:
If you make the visible seek a computed var in `SeekState` this entire function wouldn't be needed at all

Reply:
I disagree on these ones about visibleSeek because the timer that sets SeekState is different from the timer that plays the animation. It doesn't keep up with SeekState. SeekState might stay in seeking state for 1 seconds for example, but its animation should finish (`visibleSeek = nil`) after 0.3 seconds. An animation that stays in screen for 1 seconds seems little weird IMO. It's just my opinion, but I can remove it np

Reply:
It's the way YouTube does it, and it's a good way to signify to the User that the seek by tap is still active. As long as the animation is playing on the screen, the User can tap and keep seeking. If the animation disappeared, further taps won't be recognized as a seek. I think it makes a lot more sense. You can potentially lessen the value of the seek state to `0.8` or something, but I 100% think the animation should be in line with that